### PR TITLE
Update to latest cosmos-sdk

### PR DESCRIPTION
**Pull request by faddat**

### CODE REVIEW

**Comment by faddat** [comment]
I like this one.  Closes #685 among others.

File: cmd/wasmd/root.go [hunk 1]
```diff
@@ -80,8 +82,64 @@ func NewRootCmd() (*cobra.Command, app.EncodingConfig) {
...
+lru_size = 0`
+
+	return customAppTemplate, customAppConfig
+}
+
 func initRootCmd(rootCmd *cobra.Command, encodingConfig app.EncodingConfig) {
-	authclient.Codec = encodingConfig.Marshaler
 
 	rootCmd.AddCommand(
 		genutilcli.InitCmd(app.ModuleBasics, app.DefaultNodeHome),
```
Comment (RiccardoM) [inline cmd/wasmd/root.go:145]: Here you can also add the `config` command that has been brought back into v0.44: 
```go
config.Cmd()
```
Ideally you can place it after the `debug.Cmd()` entry

**Review by RiccardoM — COMMENTED** [review]
I assumed all the `third_party` proto files have been downloaded so I didn't check them in details. Aside from the small things pointed out, I also don't see the new `.pb.go` files that should be generated when the `.proto` files are updated by running `make proto-gen`

File: x/wasm/ibc.go [hunk 2]
```diff
@@ -219,58 +230,59 @@ func toWasmVMChannel(portID, channelID string, channelInfo channeltypes.Channel)
 func (i IBCHandler) OnRecvPacket(
 	ctx sdk.Context,
 	packet channeltypes.Packet,
-) (*sdk.Result, []byte, error) {
+	relayer sdk.AccAddress,
+) exported.Acknowledgement {
 	contractAddr, err := ContractFromPortID(packet.DestinationPort)
 	if err != nil {
-		return nil, nil, sdkerrors.Wrapf(err, "contract port id")
```
Comment (giansalex) [inline x/wasm/ibc.go:225]: why hide the error message?

File: x/wasm/keeper/handler_plugin.go [hunk 3]
```diff
@@ -22,12 +25,14 @@ type msgEncoder interface {
 
 // SDKMessageHandler can handles messages that can be encoded into sdk.Message types and routed.
 type SDKMessageHandler struct {
-	router   sdk.Router
-	encoders msgEncoder
+	router    sdk.Router
+	msgRouter *baseapp.MsgServiceRouter
```
Comment (giansalex) [inline x/wasm/keeper/handler_plugin.go:29]: `msgRouter` is not included in `NewSDKMessageHandler`
https://github.com/CosmWasm/wasmd/blob/e85d769ee0db346e140e49b351c9878606667ab5/x/wasm/keeper/handler_plugin.go#L54

**Comment by alpe** [comment]
Thanks @faddat  and all reviewers and contributors! This was very inspiring to see the community work on a shared goal. 👏 👏 👏 
As discussed on the community call before, I am closing this in favour of #717, now.

**Pull request closed without merging**